packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -265,22 +283,66 @@ class ScrollDragController implements Drag {
     _delegate = value;
   }
 
+  /// Determines whether to lose the existing incoming velocity when starting
+  /// the drag.
+  void _maybeLoseMomentum(double offset, Duration timestamp) {
+    if (_retainMomentum &&
+        offset == 0.0 &&
+        (timestamp == null || // If drag event has no timestamp, we lose momentum.
+         timestamp - _lastNonStationaryTimestamp > momentumRetainStationaryDurationThreshold)) {
+      // If pointer is stationary for too long, we lose momentum.
+      _retainMomentum = false;
+    }
+  }
+
+  /// If a motion start threshold exists, determine whether the threshold is
+  /// reached to start applying position offset.
+  ///
+  /// Returns false either way if there's no offset.
+  bool _breakMotionStartThreshold(double offset, Duration timestamp) {
+    if (timestamp == null) {
+      // If we can't track time, we can't apply thresholds. Should never happen.
```

Comment:
If it should never happen, assert instead?

Reply:
Clarified comment. Can be null with accessibility gestures